### ISSUE TITLE
No module named 'avalanche.benchmarks.generators'

### ISSUE THREAD

**Issue by Mming11** [Ivsucram/CDCL#1]
Thank you for this excellent work.
I encountered a problem when I reproduced your outstanding work.

train.py 71 <module>                                                                                                                                                                                             
from src.utils.data_config import resolve_data_config                                                                                                                                                            
                                                                                                                                                                                                                 
__init__.py 1 <module>                                                                                                                                                                                           
from .cdcl import *                                                                                                                                                                                              
                                                                                                                                                                                                                 
cdcl.py 3 <module>                                                                                                                                                                                               
from .utils.transformers import TransformerClassifier                                                                                                                                                            
                                                                                                                                                                                                                 
__init__.py 3 <module>                                                                                                                                                                                           
from .dataset_factory import create_dataset                                                                                                                                                                      
                                                                                                                                                                                                                 
dataset_factory.py 7 <module>                                                                                                                                                                                    
from avalanche.benchmarks.generators import nc_benchmark                                                                                                                                                         
                                                                                                                                                                                                                 
ModuleNotFoundError:                                                                                                                                                                                             
No module named 'avalanche.benchmarks.generators'

**Comment by Ivsucram** [comment]
Hi @Mming11, I will need to check it. If the Avalanche library was updated, maybe I can upload a requirements.txt or another file with the specific old version of it, and other related packages.

Thanks for testing it and raising the issue.

**Comment by Mming11** [comment]
Thank you for your quick reply. I am looking forward to your revision.

**Comment by Ivsucram** [comment]
Hi @Mming11, the problem has been fixed, although it is not a future-proof solution. A future-proof fix will be handled in a future research iteration.

Details of the fix:
The Avalanche generators function is deprecated right now. However, the authors still let the function available in their source code, via a deprecated import, so source codes based on old avalanche-lib, such as CDCL, can still run without major problems.

If you are curious about the fix, it was a single line (7) on the src/utils/data_factory.py file.

**Comment by Mming11** [comment]
Thank you for your solution, but I meet another problem when running the code.

train.py 1190 <module>
main()

train.py 578 main
num_aug_splits, mixup_active, mixup_fn) = create_loaders(

loader_factory.py 39 create_loaders
loader_source_train = create_loader(

loader.py 202 create_loader
dataset.transform = create_transform(

AttributeError:
can't set attribute

**Comment by Ivsucram** [comment]
Hi @Mming11, I have re-opened the issue.
It was my fault to not test it thoroughly. I have used a PC without a GPU and assumed that everything was right after solving the Avalance deprecation error.

I am now on a PC with GPU and I will make sure that the code executes at least a whole experiment - I will focus on MNIST<->USPS because it is a smaller problem.

**Comment by Ivsucram** [comment]
Hi @Mming11, I made sure that the code is running now.

I have uploaded two environment files:
- requirements.txt, in case you use pip
- conda-environment.yaml, in case you use conda

These files have the correct version of all packages used by me when developing the original code.

Also, I expanded the README.md file with a workaround instruction, in case you get an ImportError problem with the torchvision library.
